apps/condo/domains/organization/integrations/sbbol/sync/requestTransactions.js

```diff
@@ -124,29 +125,69 @@ async function requestTransactionsForDate ({ userId, bankAccounts, context, stat
             // when report is requested for date in future
             // when report page does not exist, for example number is out of range of available pages
-            if (get(transactions, 'error.cause') === 'WORKFLOW_FAULT') reqErrored = true
-            if (get(summary, 'error.cause') === 'WORKFLOW_FAULT') reqErrored = true
-            if (get(summary, 'error.cause') === 'DATA_NOT_FOUND_EXCEPTION') reqErrored = true
-            if (get(transactions, 'error.cause') === 'UNAUTHORIZED') {
+            if (get(transactions, 'error.cause') === SBBOL_ERRORS.WORKFLOW_FAULT) {
+                reqErrored = true
+                transactionException = SBBOL_ERRORS.WORKFLOW_FAULT
+            }
+            if (get(summary, 'error.cause') === SBBOL_ERRORS.WORKFLOW_FAULT) {
+                reqErrored = true
+                summaryException = SBBOL_ERRORS.WORKFLOW_FAULT
+            }
+
+            // DATA_NOT_FOUND_EXCEPTION means that the statement not found and cannot be generated
+            if (get(transactions, 'error.cause') === SBBOL_ERRORS.DATA_NOT_FOUND_EXCEPTION) {
+                reqErrored = true
+                transactionException = SBBOL_ERRORS.DATA_NOT_FOUND_EXCEPTION
+            }
+            if (get(summary, 'error.cause') === SBBOL_ERRORS.DATA_NOT_FOUND_EXCEPTION) {
+                reqErrored = true
+                summaryException = SBBOL_ERRORS.DATA_NOT_FOUND_EXCEPTION
+            }
+
+            // ACTION_ACCESS_EXCEPTION means that the required access parameter for the requested data
+            // is not included in the offer with the user
+            if (get(transactions, 'error.cause') === SBBOL_ERRORS.ACTION_ACCESS_EXCEPTION) {
+                reqErrored = true
+                transactionException = SBBOL_ERRORS.ACTION_ACCESS_EXCEPTION
+            }
+            if (get(summary, 'error.cause' === SBBOL_ERRORS.ACTION_ACCESS_EXCEPTION)) {
+                reqErrored = true
+                summaryException = SBBOL_ERRORS.ACTION_ACCESS_EXCEPTION
+            }
+
+            if (get(transactions, 'error.cause') === SBBOL_ERRORS.UNAUTHORIZED) {
                 allDataReceived = false
                 reqErrored = false
                 accessTokenIndex++
                 if (accessTokenIndex < accessTokens.length) {
                     sbbolFintechClient = initSbbolClientWithToken(accessTokens[accessTokenIndex])
                 } else {
                     reqErrored = true
+                    transactionException = SBBOL_ERRORS.UNAUTHORIZED
+                    summaryException = SBBOL_ERRORS.UNAUTHORIZED
                 }
             }
         }
 
         accessTokenIndex = 0
 
         if (summary) {
+            const meta = {}
+            if (transactionException || summaryException) {
+                meta.lastSyncStatus = BANK_SYNC_TASK_STATUS.ERROR
+                meta.transactionException = transactionException ?? null
```

Comment:
Is that a right place to store task errors? For me it would be better to save that to task meta or integration context